src/telemetry/Utils.cc

```diff
@@ -0,0 +1,54 @@
+#include "Utils.h"
+
+#include "zeek/ID.h"
+#include "zeek/Reporter.h"
+#include "zeek/Val.h"
+#include "zeek/telemetry/telemetry.bif.h"
+#include "zeek/util.h"
+
+using namespace zeek;
+
+namespace zeek::telemetry::detail {
+
+std::string BuildFullPrometheusName(std::string_view prefix, std::string_view name, std::string_view unit,
+                                    bool is_sum) {
+    if ( prefix.empty() || name.empty() )
+        reporter->FatalError("Telemetry metric families must have a non-zero-length prefix and name");
+
+    std::string fn = util::fmt("%s_%s", prefix.data(), name.data());
+    std::for_each(fn.begin(), fn.end(), [](char& c) {
+        if ( ! std::isalnum(c) )
+            c = '_';
+    });
+
+    // We were previously using "1" to mean "no unit value" for whatever reason, so we have to handle that now
+    // to mean the same thing.
+    if ( ! unit.empty() && unit != "1" )
+        fn.append("_").append(unit);
+
+    if ( is_sum )
```

Comment:
I was referring to this in my earlier comment — looks like this is the only actual use of `is_total`/`is_sum` now.